internal/converter/internal/otelcolconvert/converter_lokireceiver.go

```diff
@@ -0,0 +1,143 @@
+	"github.com/grafana/agent/internal/component/loki/source/api"
+	"github.com/grafana/agent/internal/component/otelcol"
+	otel_loki "github.com/grafana/agent/internal/component/otelcol/receiver/loki"
+	"github.com/grafana/agent/internal/converter/diag"
+	"github.com/grafana/agent/internal/converter/internal/common"
+	"github.com/grafana/agent/internal/converter/internal/promtailconvert/build"
+	"github.com/grafana/dskit/server"
+	"github.com/grafana/loki/clients/pkg/promtail/scrapeconfig"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/lokireceiver"
+	"go.opentelemetry.io/collector/component"
+)
+
+func init() {
+	converters = append(converters, lokiReceiverConverter{})
+}
+
+type lokiReceiverConverter struct{}
+
+func (lokiReceiverConverter) Factory() component.Factory { return lokireceiver.NewFactory() }
+
+func (lokiReceiverConverter) InputComponentName() string { return "" }
+
+func (lokiReceiverConverter) ConvertAndAppend(state *State, id component.InstanceID, cfg component.Config) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	label := state.FlowComponentLabel()
+
+	otelArgs := toOtelcolReceiverLoki(state, id)
+	otelBlock := common.NewBlockWithOverride([]string{"otelcol", "receiver", "loki"}, label, otelArgs)
+
+	logsReceivers := []loki.LogsReceiver{common.ConvertLogsReceiver{
+		Expr: StringifyBlock(otelBlock) + ".receiver",
+	}}
+	apiArgs := toLokiSourceApi(logsReceivers, cfg.(*lokireceiver.Config))
+	apiBlock := common.NewBlockWithOverride([]string{"loki", "source", "api"}, label, apiArgs)
+
+	diags.Add(
+		diag.SeverityLevelInfo,
+		fmt.Sprintf("Converted %s into %s", StringifyInstanceID(id), StringifyBlock(otelBlock)),
+	)
+
+	diags.Add(
+		diag.SeverityLevelInfo,
+		fmt.Sprintf("Converted %s into %s", StringifyInstanceID(id), StringifyBlock(apiBlock)),
+	)
+
+	// Do this at the end in reverse order so the pipeline looks a little better
+	state.Body().AppendBlock(apiBlock)
+	state.Body().AppendBlock(otelBlock)
+	return diags
+}
+
+func toOtelcolReceiverLoki(state *State, id component.InstanceID) *otel_loki.Arguments {
+	var (
+		nextLogs = state.Next(id, component.DataTypeLogs)
+	)
+
+	return &otel_loki.Arguments{
+		Output: &otelcol.ConsumerArguments{
+			Logs: ToTokenizedConsumers(nextLogs),
+		},
+	}
+}
+
+func toLokiSourceApi(logsReceivers []loki.LogsReceiver, cfg *lokireceiver.Config) *api.Arguments {
+	ptc := &scrapeconfig.PushTargetConfig{
+		Server:        toServer(&cfg.Protocols),
+		Labels:        nil,
+		KeepTimestamp: cfg.KeepTimestamp,
+	}
+
+	args := build.ToLokiApiArguments(ptc, logsReceivers)
+	return &args
+}
+
+func toServer(cfg *lokireceiver.Protocols) server.Config {
+	return server.Config{
+		HTTPListenAddress:               "",
+		HTTPListenPort:                  0,
+		HTTPConnLimit:                   0,
+		HTTPServerReadTimeout:           0,
+		HTTPServerWriteTimeout:          0,
+		HTTPServerIdleTimeout:           0,
+		GRPCListenAddress:               "",
```

Comment:
otel supports GRPC for their receiver and while ours has them wired up as flow arguments it is not supported. I created a new issue to add this support here: https://github.com/grafana/agent/issues/6783